### PR TITLE
Rewrite from scratch for v0.6

### PR DESCRIPTION
**Pull request by gaika**
Just a heads up 

- [x] Generate wrap.jl that exports all ArrayFire functions safely including graphics, sparse, etc.
- [x] Compile level type and dimensions inference matching native Julia `Array` behaviour
- [x] Module initialization and precompilation
- [x] Support for Julia v0.6 broadcast fusion
- [x]  Arithmetic operations as operators on arrays
- [ ] Random number engines
- [x] Call gc() and device_gc() on device memory pressure
- [x] `@afgc` to immediately release references to temporary arrays going out of scope without waiting for gc()
- [ ] Backend / card selection wrappers
- [x]  Indexing of arrays
- [x] Graphics functions (`figure`, `plot`)
- [x] Julia sparse to ArrayFire sparse conversion functions
- [ ] Type inference for all functions
- [x] Julia v0.5
- [x] Julia v0.6
- [ ] Julia v0.7 nightly
- [ ] more examples from ArrayFire?

Comments, suggestions, PRs are welcome.

### CODE REVIEW

**Comment by oschulz** [comment]
Why not use the prefix `af_` in wrap.jl again, as before? Won't we run into trouble with functions like `fft` that clash with `Base.fft`?

**Comment by oschulz** [comment]
I'd like to contribute to this - could we turn this into a branch, so that people can make PRs?

**Comment by aviks** [comment]
You can do a PR against https://github.com/gaika/ArrayFire.jl/

Go here: https://github.com/gaika/ArrayFire.jl/compare/master...Godisemo:master and change the  fork repo as appropriate

**Comment by oschulz** [comment]
Will do!

File: src/util.jl [hunk 1]
```diff
@@ -5,9 +5,23 @@ import Base: RefValue, @pure, display, show, clamp, find, cumsum, cumprod, cummi
 export constant, select, get_last_error, err_to_string, sort_index
 export mean_weighted, var_weighted, set_array_indexer, set_seq_param_indexer
 
-function afgc(threshold = 4e9)
+global const af_threshold = Ref(4*1024*1024*1024)
```
Comment (musm) [inline src/util.jl]: This is already in global scope, is it  necessary to declare it a `global const` vs just a  `const`?

Comment (gaika) [inline src/util.jl]: thanks!

**Comment by oschulz** [comment]
I was thinking - maybe it's time to merge this? Many users will have moved to Julia v0.6 now, and better an ArrayFire that may still be lacking some minor things, than now 0.6-compatible ArrayFire at all? :-)

**Comment by ranjanan** [comment]
I agree. @gaika you should have received an invite for write access. Thanks so much for this PR once more! :-)

**Comment by oschulz** [comment]
Thanks a lot, @gaika!

**Pull request merged**